sims_GCRCatSimInterface_withlensing

mpwiesner opened the pull request.

Commit:
Added Nan's lensing files

Added a list of all the FITS image to the data folder and began editing InstanceCatalogWriter.py

Changed to python3 print.

First attempt at hooking up InstanceCatalogWriter to host images.

Made changes and tested on jupyter-dev.

Removed .ds_Store files

Responding to requested changes in generate_lensed_hosts_***.py, InstanceCatalogWriter.py and HostImages.py

One more edit to README.lensing

Removed .DS_Store and added to .gitignore.

Modified after testing on jupyter-dev.

Cleaned out second data folder.

mpwiesner commented:
Includes edits to now produce and then include FITS images of lensed hosts.

jbkalmbach commented:
I have a copy of the lensed images in `/global/cscratch1/sd/brycek/outputs/`

jchiang87 commented:
I'll have a closer look at this over the weekend.  In the meantime, can we remove the `.DS_store` files please?

mpwiesner commented:
@jchiang87 I removed the files.

jbkalmbach commented:
Awesome. Thanks @jchiang87

jchiang87 commented:
@jbkalmbach I'm finally getting around to looking at this.  Can you please give me access to the lensed image directory:
```
[cori03] ls /global/cscratch1/sd/brycek/outputs/
ls: cannot access '/global/cscratch1/sd/brycek/outputs/': Permission denied
```

jchiang87 commented on python/desc/sims/GCRCatSimInterface/data/generate_lensed_hosts_agn_final.py:
```diff
@@ -0,0 +1,212 @@
+%matplotlib inline
```
This file looks very similar to `python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py`.  Perhaps it was committed by mistake?   If so, please remove it.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,223 @@
+
+if __name__ == '__main__':
+
+    from tqdm import tqdm_notebook
```
We should try to avoid introducing dependencies on third party packages unless they are really needed.  Can you rewrite this without using `tqdm`?

jbkalmbach commented:
Ok, I changed everything to `lsst` group. Let me know if that didn't work for you.

jchiang87 commented:
Still no luck.  I think you may need to do `chmod o+rx /global/cscratch1/sd/brycek` in order for people in the `lsst` group to see any subfolders in your scratch area.

jbkalmbach commented:
Ok, I ran that command. I hope it works now otherwise let me know if there's another place I can put them that would make it easy.

jchiang87 commented:
Thanks, I can access that area now.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,223 @@
+#matplotlib inline
+#config InlineBackend.figure_format = 'retina'
+
+import numpy as np
```
add
```
import os
```
See below.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,223 @@
+
+    lensed_mag_b, lensed_image_b = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_b,lens_P)
+
+    fits_limg_b = "./outputs/agn_lensed_bulges/" + str(lens_P['UID_lens']) + "_" + str(lensed_mag_b) + "_bulge.fits" #\
```
I think you need to create the output directory before you can run `pyfits.writeto`, e.g., add a line
```
    os.makedirs('./outputs/agn_lensed_bulges', exist_ok=True)
```

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,223 @@
+
+    lensed_mag_d, lensed_image_d = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_d,lens_P)
+
+    fits_limg_d = "./outputs/agn_lensed_disks/" + str(lens_P['UID_lens']) + "_" + str(lensed_mag_d) + "_disk.fits"#\
```
As with the agn_lensed_bulges, I think you should add
```
    os.makedirs('./outputs/agn_lensed_disks', exist_ok=True)
```
before this line.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,223 @@
+    print(ahb.index)
+            
+#     for i in tqdm_notebook(ahb.index, desc="Main Loop"):
+    for i in tqdm_notebook(ahb.index[:1000], desc="Main Loop"):
```
Do you really want to cut this off at 1000?

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_sne.py:
```diff
@@ -0,0 +1,340 @@
+
+    lensed_mag_b, lensed_image_b = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_b,lens_P)
+
+    fits_limg_b = "./outputs/sne_lensed_bulges/" + str(lens_P['UID_lens']) + "_" + str(lensed_mag_b) + "_bulge.fits"#\
```
Insert
```
    os.makedirs('./output/sne_lensed_bulges', exist_ok=True)
```
before this line.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_sne.py:
```diff
@@ -0,0 +1,340 @@
+
+    lensed_mag_d, lensed_image_d = lensed_sersic_2d(xi1,xi2,yi1,yi2,srcP_d,lens_P)
+
+    fits_limg_d = "./outputs/sne_lensed_disks/" + str(lens_P['UID_lens']) + "_" + str(lensed_mag_b) + "_disk.fits" #\
```
Insert
```
    os.makedirs('./outputs/sne_lensed_disks', exist_ok=True)
```

jchiang87 commented on python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:
```diff
@@ -180,6 +182,8 @@ def __init__(self, opsimdb, descqa_catalog, dither=True,
                 raise IOError("Path to Proto DC2 AGN database does not exist.")
 
         self.sprinkler = sprinkler
+        self.host_image_dir = host_image_dir
```
Since the code will crash (frustratingly after running for many minutes) if the default values of `None` are used for `host_image_dir` and `host_data_dir`, you should put a guarding `if ... is None` check here and raise an exception if the defaults are used.  For example, see how this is done for the `agn_db_name` parameter.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:
```diff
@@ -116,7 +117,8 @@ class InstanceCatalogWriter(object):
                  min_mag=10, minsource=100, proper_motion=False,
                  imsim_catalog=False, protoDC2_ra=0, protoDC2_dec=0,
-                 agn_db_name=None, sprinkler=False):
+                 agn_db_name=None, sprinkler=False, host_image_dir=None,
```
The main script `generateInstCat.py` also needs to be modified to provide options to specify `host_image_dir` and `host_data_dir`.  The code as written will not work since there is no place for the user to set these arguments.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:
```diff
@@ -325,6 +329,21 @@ def write_catalog(self, obsHistID, out_dir='.', fov=2):
+            host_cat = hostImage(obs_md.pointingRA, obs_md.pointingDec, fov)
+            host_cat.write_host_cat(os.path.join(self.host_image_dir, 'agn_lensed_bulges'),
+                                    os.path.join(self.host_data_dir, 'agn_host_bulge.csv.gz'),
+                                    sprinkled_host_name)
```
Each of these cases of `sprinkled_host_name` should instead be `os.path.join(outdir, sprinkled_host_name)`.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:
```diff
@@ -212,7 +216,7 @@ def write_catalog(self, obsHistID, out_dir='.', fov=2):
         gal_name = 'gal_cat_%d.txt' % obsHistID
         knots_name = 'knots_cat_%d.txt' % obsHistID
-        #agn_name = 'agn_cat_%d.txt' % obshistid
+        sprinkled_host_name = 'spr_hosts_%d.txt' % obsHistID
```
This filename should be added to the `object_catalogs` list below.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/README.lensing:
```diff
@@ -0,0 +1,11 @@
+To generate the FITS files of the lensed hosts, you will need to run the following commands only once:
```
I think these instructions need more details.  First off, the `load_in_data_[sne,agn]` functions assume that these functions will be run from the `python/desc/sims/GCRCatSimInterface` folder, otherwise the `pd.read_csv` calls would fail.  However, I'm not sure this folder is the best place to put those data since there is already a `data` folder in this repo.  Second, the code is written so that the generated FITS images will be written to `python/desc/sims/GCRCatSimInterface/outputs`, which similarly may not be the best place.    I would move the data files to the top level `data` folder, and make it so that the user can specify where the output images should be written.

danielsf replied:
I agree with Jim that the default directory for these FITS images should be `sims_GCRCatSimInterface/data/`

jchiang87 commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,76 @@
+        else:
+            raise ValueError('Invalid Galaxy Component Type in filename')
+
+        cat_str = 'object %i %f %f %f %s %f 0 0 0 0 0 %s 0.01 0\n' % (sys_id,
```
These object entries for the instance catalog are missing the rest frame and Galactic extinction parameters.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,76 @@
+                                                                      df_line['raPhoSim_lens'],
+                                                                      df_line['decPhoSim_lens'],
+                                                                      sys_magNorm,
+                                                                      df_line['sedFilepath'],
```
Since the code is letting the user specify the image directory (which they may choose to relocate for staging purposes in the simulation pipeline), it seems like this should be `os.path.basename(df_line['sedFilepath'])` so that the image directory can be specified by the user when running phosim.

jchiang87 reviewed (changes requested):
I've done my initial pass through these changes and ran into a number of issues in trying to run this code, so I have several requests for changes.

mpwiesner commented:
@jchiang87 I will work on these changes this week.

jbkalmbach commented:
Hi @mpwiesner. Any progress on this? Do you want a hand with it?

mpwiesner commented:
Hi @jbkalmbach, apologies for the slow response.  I have made some progress, but I am not done yet.  I will try to get this done by the end of the week.

mpwiesner commented:
I believe I have addressed all the changes requested.  @jbkalmbach if you get a chance, could you look at the edits to ` HostImages.py `?  I added entries for rest frame and galactic extinction parameters, but I'm not quite sure if I need to point it to where to find those parameters in the data files.

jbkalmbach commented:
Hey @mpwiesner I'm looking at it and trying to run the code at NERSC now. One comment just looking over the files list. Since `sims_GCRCatSimInterface` imports Twinkles already, do you think we could not reproduce some of the files you've added in the data directory since they are already in the Twinkles data directory. We could use the `TWINKLES_DIR` environment variable like we did here: https://github.com/jbkalmbach/sims_GCRCatSimInterface/blob/master/python/desc/sims/GCRCatSimInterface/TwinklesClasses.py#L15

jbkalmbach commented:
Hi @mpwiesner, I made some changes on a fork of your branch and submitted a PR to you. When you have time take a look and see if everything looks right and then once merged those changes should show up here.

mpwiesner commented:
Great, thanks @jbkalmbach.

jbkalmbach commented:
@jchiang87 I think this is ready for another look.

danielsf commented on bin.src/generateInstCat.py:
```diff
@@ -8,6 +8,10 @@
                     help='path to the OpSim database to query')
 parser.add_argument('--agn_db_name', type=str,
                     help='File of AGN parameters generated by create_agn_db.py')
+parser.add_argument('--host_image_dir', type=str,
```
Please add default values for `host_image_dir` and `host_data_dir`.  Either set them to `None` (in which case, I assume, the InstanceCatalogs will not contain FITS profiles), or point them to actual default directories that the user can be guaranteed will exist.

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,232 @@
+#matplotlib inline
```
We have been putting the support scripts (e.g. `create_agn_db.py`) into `bin.src/`.  I think that is where the `generate_lensed_hosts*.py` scripts should also go.

danielsf commented on python/desc/sims/GCRCatSimInterface/README.lensing:
```diff
@@ -0,0 +1,16 @@
+To generate the FITS files of the lensed hosts, you will need to run the following commands only once:
```
I think this `README` file should be moved to the top-level `sims_GCRCatSimInterface/` directory, so that it is more visible.

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,232 @@
+import scipy.special as ss
+import om10_lensing_equations as ole
+
+outdir = "./outputs/"
```
Please make the output directory a command line argument of this script.  That way, we can redirect the output to some shared project storage space without having to edit source code.

Do the same for the `generate_lensed_hosts_sne.py` script.

You will, of course, need to edit `README.lensing` accordingly.

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,232 @@
+import om10_lensing_equations as ole
+
+outdir = "./outputs/"
+data_dir = os.path.join(os.path.abspath("../../../../"),'data')
```
You can direct to the `data` directory in a location-independent way using
```
from lsst.utils import getPackageDir
data_dir = os.path.join(getPackageDir('sims_GCRCatSimInterface'), 'data')
```
or, if you don't want to depend on `lsst.utils`
```
data_dir = os.path.join(os.environ['SIMS_GCRCATSIMINTERFACE_DIR'], 'data')
```
`SIMS_GCRCATSIMINTERFACE_DIR` will be set by EUPS when sims_GCRCatSimInterface is setup.

danielsf commented:
@mpwiesner 

When I just ran `generate_lensed_hosts_agn.py` as is, I got
```
Traceback (most recent call last):
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 868, in _free_space_check
    yield
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/base.py", line 659, in _writeto
    self._writeto_internal(fileobj, inplace, copy)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/base.py", line 665, in _writeto_internal
    data_offset, data_size = self._writedata(fileobj)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/base.py", line 595, in _writedata
    size += self._writedata_internal(fileobj)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/image.py", line 641, in _writedata_internal
    fileobj.writearray(output)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/file.py", line 316, in writearray
    _array_to_file(array, self._file)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 674, in _array_to_file
    return write(arr, outfile)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 652, in <lambda>
    write = lambda a, f: a.tofile(f)
OSError: 1000000 requested and 0 written

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 868, in _free_space_check
    yield
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/hdulist.py", line 886, in writeto
    hdu._writeto(hdulist._file)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/base.py", line 659, in _writeto
    self._writeto_internal(fileobj, inplace, copy)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 885, in _free_space_check
    raise OSError(error_message + str(exc))
OSError: 1000000 requested and 0 written

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "generate_lensed_hosts_agn.py", line 232, in <module>
    generate_lensed_host(xi1, xi2, lensP, srcPb, srcPd)    
  File "generate_lensed_hosts_agn.py", line 198, in generate_lensed_host
    pyfits.writeto(fits_limg_d, lensed_image_d.astype("float32"), overwrite=True)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/utils/decorators.py", line 507, in wrapper
    return function(*args, **kwargs)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/convenience.py", line 433, in writeto
    checksum=checksum)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/utils/decorators.py", line 507, in wrapper
    return function(*args, **kwargs)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/base.py", line 377, in writeto
    checksum=checksum)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/utils/decorators.py", line 507, in wrapper
    return function(*args, **kwargs)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/hdu/hdulist.py", line 887, in writeto
    hdu._postwriteto()
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/contextlib.py", line 99, in __exit__
    self.gen.throw(type, value, traceback)
  File "/global/common/software/lsst/common/miniconda/py3-4.3.21/lib/python3.6/site-packages/astropy/io/fits/util.py", line 885, in _free_space_check
    raise OSError(error_message + str(exc))
OSError: 1000000 requested and 0 written
```

danielsf commented:
I was using the
```
/global/common/software/lsst/cori-haswell-gcc/Run1.2p_setup.bash
```
stack on Cori

danielsf commented:
Actually, this might just mean my home directory is full.  Let me free up some space and try again

danielsf commented:
Sorry.  It was just me

mpwiesner commented:
Ok, thanks for checking @danielsf

danielsf commented:
When I ran `generateInstanceCatalog.py`, `spr_hosts_*.txt` came out looking like
```
object 30128609804385 52.950465 -27.557474 17.450824 galaxySED/Exp.32E09.1Z.spec.gz 0.620000 0 0 0 0 0 Exp.32E09.1Z.spec.gz 
0.01 0 CCM 0.610313 1.835133 CCM 0.023670 3.100000
object 30378476044385 52.948355 -27.574379 17.336021 galaxySED/Exp.25E09.1Z.spec.gz 0.620000 0 0 0 0 0 Exp.25E09.1Z.spec.gz 
0.01 0 CCM 0.501649 3.612045 CCM 0.023138 3.100000
```
It looks like the `sedFilepath` is getting copied into where the FITS file name should go.  Is that the correct behavior?

danielsf commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,81 @@
+
+
+    def format_catalog(self, df_line, fits_file_name, image_dir):
+
```
Please add a docstring to this method explaining what it does and what each of the arguments are.

danielsf commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,81 @@
+        return cat_str
+
+    def write_host_cat(self, image_dir, input_cat, output_cat, append=False):
+
```
Please add a docstring to this method explaining what it does and what each of its arguments are.

danielsf commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,81 @@
+        image_ids = np.array([image_name.split('_')[0] for image_name in image_list], dtype=np.int)
+        
+        for sys_ra, sys_dec in zip(host_df['raPhoSim_lens'], host_df['decPhoSim_lens']):
+            ang_sep_list.append(angularSeparation(sys_ra, sys_dec, self.ra, self.dec))
```
`angularSeparation` should be able to just take a numpy arrays, i.e.
```
ang_sep = angularSeparation(host_df['raPhoSim_lens'], host_df['decPhoSim_lens'], self.ra, self.dec)
```
should return what you want without having to create a list and then cast it as a numpy array.

mpwiesner commented:
No.  I'll take a look to see where the bug is.

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,232 @@
+twinkles_data_dir = os.path.join(os.environ['TWINKLES_DIR'], 'data')
+
+def load_in_data_agn():
+
```
Please add docstrings to all of the methods in these scripts.  It will make them easier to maintain/debug in the future.

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_sne.py:
```diff
@@ -0,0 +1,346 @@
+
+def random_location(Reff_src, qs, phs, ns):
+    """Sample a random (x, y) location from the surface brightness
+    profile of the galaxy."""
```
Please define what the arguments of this method should be.

danielsf commented on python/desc/sims/GCRCatSimInterface/om10_lensing_equations.py:
```diff
@@ -0,0 +1,168 @@
+
+vc = 2.998e5 #km/s
+G = 4.3011790220362e-09 # Mpc/h (Msun/h)^-1 (km/s)^2
+apr =  206269.43        # 1/1^{''}
```
Is this 1/arcsec in radians?  I get

`3600*180/pi ~ 206264.8`

danielsf commented on python/desc/sims/GCRCatSimInterface/om10_lensing_equations.py:
```diff
@@ -0,0 +1,168 @@
+data_dir = os.path.join(os.path.abspath("../../../../"),'data')
+
+def Dc(z):
+    res = p15.comoving_distance(z).value*p15.h
```
I don't think we are actually using the Planck cosmology for the extragalactic catalogs.  The GCR allows you to get access to the cosmology we are using from something like
```
cat = GCRCatalogs.load_catalogs(yaml_file)
cosmology = cat.cosomology
```
(see, for instance
https://github.com/LSSTDESC/sims_GCRCatSimInterface/blob/master/python/desc/sims/GCRCatSimInterface/CatalogClasses.py#L297 )

You can get access to the GCR catalo though any of the `*DESCQAObject`s instantiated in the `InstanceCatalogWriter`, for example, here

https://github.com/LSSTDESC/sims_GCRCatSimInterface/blob/master/python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py#L274

danielsf commented on python/desc/sims/GCRCatSimInterface/generate_lensed_hosts_sne.py:
```diff
@@ -0,0 +1,346 @@
+
+
+def check_random_locations():
+    
```
Please add docstrings for the methods in this script.  It will make it easier to maintain this code in the future.

danielsf reviewed (changes requested):
As I said in the broader `conversation`, I think there is a bug in that, when I produce InstanceCatalogs, the column that should refer to FITS profiles contains a version of the SED file name.

In addition to that, I think the code should be reorganized.  The scripts for generating the host profiles should be moved into `bin.src/` with the other executable scripts, and the resulting outputs should be more easily redirected by the user, since we are going to want to generate these FITS profiles in some global project space without editing source code.

Also: when it does come time to merge this PR, I suspect there will be a nasty rebase.  `generateInstanceCatalogs.py` has gone through a lot of changes since this PR was originally reopened.  I would recommend doing a rebase to `master` and making sure that nothing breaks.  I did about 2/3 of the offending commits to `generateInstanceCatalog.py`, so if you want me to do the rebase, I can.

danielsf commented:
I think the offending line as far as the SED file names appearing in the place of FITS profiles is here

https://github.com/mpwiesner/sims_GCRCatSimInterface/blob/master/python/desc/sims/GCRCatSimInterface/HostImages.py#L41

danielsf commented:
If it helps, this is the bash script I used to generate InstanceCatalogs when testing the PR
```
output_dir=workspace/catalogs/

python bin.src/generateInstCat.py --db /global/projecta/projectdirs/lsst/groups/
SSim/DC2/minion_1016_desc_dithered_v4.db --agn_db_name /global/projecta/projectd
irs/lsst/groups/SSim/DC2/agn_db_mbh_7.0_m_i_30.0.sqlite --descqa_catalog proto-d
c2_v4.6.1_test --fov 0.2 --out_dir ${output_dir} --protoDC2_ra 55.064 --protoDC2
_dec -29.783 --enable_sprinkler --ids 2443890 --host_data_dir data/  --host_imag
e_dir python/desc/sims/GCRCatSimInterface/outputs/
```

Commit:
Responding to requested changes, mostly to lensing routines

Commit:
Fixed a few bugs in HostImages and InstanceCatalogWriter

Commit:
Fixed several errors I had made in InstanceCatalogWriter.py and HostImages.py. Also updated comments to match the format used in other files.

mpwiesner commented:
Here is the file I used to generate instance catalogs on Cori (slightly modified from Scott's above): 
```
output_dir=workspace/catalogs/

python sims_GCRCatSimInterface/bin.src/generateInstCat.py --db /global/projecta/projectdirs/lsst/groups/\
SSim/DC2/minion_1016_desc_dithered_v4.db --agn_db_name /global/projecta/projectd\
irs/lsst/groups/SSim/DC2/agn_db_mbh_7.0_m_i_30.0.sqlite --descqa_catalog proto-d\
c2_v4.6.1_test --fov 0.2 --out_dir /global/homes/m/mwiesner/sims_GCRCatSimInterface/workspace/catalogs/today\
 --protoDC2_ra 55.064 --protoDC2\
_dec -29.783 --enable_sprinkler --ids 2443890 --host_data_dir /global/homes/m/mwiesner/sims_GCRCatSimInterface/data\
 --host_image_dir /global/homes/m/mwiesner/sims_GCRCatSimInterface/data/outputs
```

mpwiesner commented:
@danielsf @jchiang87 @jbkalmbach I have updated the code and verified that it is correctly producing instance files that include static lenses.  Could you take a look when you have a chance?

mpwiesner committed:
updating generateInstCat.py

jbkalmbach committed:
Removed some old files. Updated README with info on data files. Cleaned up generate host files a little.

mpwiesner committed:
correcting error in last commit

mpwiesner committed:
Merge branch 'master' of https://github.com/mpwiesner/sims_GCRCatSimInterface

mpwiesner committed:
removing additional line in generateInstCat.py

mpwiesner committed:
Edited 'README.lensing'

mpwiesner committed:
Removed file called :.  Made some important edits to InstanceCatalogWriter.py. Previously it was producing 3 galaxy catalogs (agn_gal*, bulge_gal*, disk_gal*) but only listing gal_* in the phosim instance file.  This was fixed

mpwiesner committed:
Removed file called 'None'.

jbkalmbach committed:
Changed default dir again.

jbkalmbach committed:
Resolved merge conflicts.

jbkalmbach commented:
I used this setup script:

```
output_dir=workspace/catalogs/

python bin.src/generateInstCat.py --db /global/projecta/projectdirs/lsst/groups/SSim/DC2/minion_1016_desc_dithered_v4.db --agn_db_name /global/projecta/projectdirs/lsst/groups/SSim/DC2/agn_db_mbh_7.0_m_i_30.0_gcr_protodc2_v3.db --descqa_catalog proto-dc2_v3.0 --fov 0.2 --out_dir ${output_dir} --protoDC2_ra 55.064 --protoDC2_dec -29.783 --enable_sprinkler --ids 2443890 --host_data_dir data/  --host_image_dir data/outputs/
```

to make sure the AGN appear because Twinkles has cached protoDC2 v3 galaxy ids.

mpwiesner committed:
Merge pull request #4 from jbkalmbach/master

Minor changes

danielsf commented on python/desc/sims/GCRCatSimInterface/InstanceCatalogWriter.py:
```diff
@@ -1,590 +1,647 @@
+        written_catalog_names.append(star_name)
+
+        # TODO: Find a better way of checking for catalog type
+        if 'knoimsimts' in self.descqa_catalog:
```
`knoimsimts` is probably not actually a thing ;)

jchiang87 commented on bin.src/generate_lensed_hosts_agn.py:
```diff
@@ -0,0 +1,224 @@
+import numpy as np
```
Since this script, `generate_lensed_hosts_sne.py`, and `om10_lensing_equations.py` are in the `bin.src` folder, they should have the `#!/usr/bin/env python` directive as the first line.  This way the `scons` command will put that executable in the `bin` folder and the scripts will be included in the PATH.

jchiang87 commented on python/desc/sims/GCRCatSimInterface/HostImages.py:
```diff
@@ -0,0 +1,100 @@
+                                                                      sys_magNorm,
+                                                                      df_line['sedFilepath'],
+                                                                      df_line['redshift'],
+                                                                      str(lens_id) + "_" + str(sys_magNorm) + "_" + str(gal_type) + ".fits",
```
I'm wondering if this file specification for the FITS image files for the lensed object should also prepend the subfolder name, e.g., `agn_lensed_bulges`.  My understanding is that phosim expects a single `--imageDir` path for these sorts of files, so the instance catalog entry needs to be a correct relative path to that `imageDir` value.

jchiang87 reviewed (changes requested):
In addition to the two inline change requests, we should also fix the whitespace issue with `InstanceCatalogWriter.py` so that a proper rebase can be done (and to preserve the file-level history).

jbkalmbach commented:
Any complaints if I close this since we merged #56 ?

danielsf commented:
Oh yeah.  I forgot this was even open.
